File: app/src/main/java/com/innovation4you/napking/util/platform/OneTimeLayoutChangeListener.java
package com.innovation4you.napking.util.platform;

import android.view.View;

public abstract class OneTimeLayoutChangeListener implements View.OnLayoutChangeListener {

	@Override
	public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
		v.removeOnLayoutChangeListener(this);
		onLayoutChange();
	}

	public abstract void onLayoutChange();
}

File: app/src/main/java/com/innovation4you/napking/util/platform/OneTimeLayoutChangeListener.kt
package com.innovation4you.napking.util.platform

import android.view.View

abstract class OneTimeLayoutChangeListener : View.OnLayoutChangeListener {

    override fun onLayoutChange(v: View, left: Int, top: Int, right: Int, bottom: Int, oldLeft: Int, oldTop: Int, oldRight: Int, oldBottom: Int) {
        v.removeOnLayoutChangeListener(this)
        onLayoutChange()
    }

    abstract fun onLayoutChange()
}
